packages/truffle-external-compile/index.js

```diff
@@ -57,15 +57,23 @@ const compile = callbackify(async function(options) {
 
   const contracts = {};
   for (let target of targets) {
-    const pattern = path.join(cwd, target.path);
-
-    for (let preprocessed of await glob( pattern, { follow: true })) {
-      debug("processing target: %s", preprocessed);
-      const input = fs.readFileSync(preprocessed).toString();
-
-      const output = (target.command)
-        ? execSync(target.command, { cwd, input })
-        : input;
+    expect.one(target, [ "path", "command" ]);  // also allows both
+
+    if (target.path != undefined) {
```

Comment:
this is safer to test for undefined: `typeof target.path !== "undefined")`

Reply:
Yep, except this is on purpose to allow `null` as a valid non-value :)